src/Titon/Route/Router.hh

```diff
@@ -637,7 +636,7 @@ class Router implements Subject {
             }
 
             /** @var \Titon\Route\Route $newRoute */
-            $newRoute = Registry::factory($class, [$newPath, static::buildAction($newAction)], false);
+            $newRoute = Depository::getInstance()->make($class, $newPath, static::buildAction($newAction));
```

Comment:
I'm not sure we should use the `Depository` here as the constructor arguments are already explicit and `Registry::factory()` simply instantiates a class using reflection. It also adds an unnecessary dependency to the context package.